infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/KubernetesProvider.java

```diff
@@ -0,0 +1,221 @@
+import org.wso2.testgrid.common.config.DeploymentConfig;
+import org.wso2.testgrid.common.config.InfrastructureConfig;
+import org.wso2.testgrid.common.config.Script;
+import org.wso2.testgrid.common.exception.CommandExecutionException;
+import org.wso2.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.testgrid.common.util.DataBucketsHelper;
+import org.wso2.testgrid.common.util.StringUtil;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static java.nio.file.StandardOpenOption.APPEND;
+import static java.nio.file.StandardOpenOption.CREATE;
+
+/**
+ * This class provide infrastructure for the kubernetes architecture.
+ */
+public class KubernetesProvider implements InfrastructureProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(KubernetesProvider.class);
+    private static final String KUBERNETES_PROVIDER = "KUBERNETES";
+
+    @Override
+    public String getProviderName() {
+        return KUBERNETES_PROVIDER;
+    }
+
+    @Override
+    public boolean canHandle(Script.ScriptType scriptType) {
+        return scriptType == Script.ScriptType.KUBERNETES;
+    }
+
+    @Override
+    public void init(TestPlan testPlan) throws TestGridInfrastructureException {
+
+    }
+
+    @Override
+    public void cleanup(TestPlan testPlan) throws TestGridInfrastructureException {
+
+    }
+
+    /**
+     * This method invoke the infrastructure creation script which would create the infrastructure through
+     * kubectl
+     *
+     * @param testPlan {@link TestPlan} with current test run specifications
+     * @param script with the details of the infra creation script specifications
+     * @return Returns the InfrastructureProvisionResult
+     * @throws TestGridInfrastructureException when there is an error in the script
+     */
+    @Override
+    public InfrastructureProvisionResult provision(TestPlan testPlan, Script script)
+            throws TestGridInfrastructureException {
+        String testPlanLocation = Paths.get(testPlan.getInfrastructureRepository()).toString();
+        InfrastructureConfig infrastructureConfig = testPlan.getInfrastructureConfig();
+        setAccessKeyFileLocation(testPlan);
+        setProperties(testPlan);
+        logger.info("Executing provisioning scripts...");
+        try {
+            ShellExecutor executor = new ShellExecutor(null);
+            InfrastructureProvisionResult result = new InfrastructureProvisionResult();
+            String infraInputsLoc = DataBucketsHelper.getInputLocation(testPlan)
+                    .toAbsolutePath().toString();
+
+            String infraOutputsLoc = DataBucketsHelper.getOutputLocation(testPlan)
+                    .toAbsolutePath().toString();
+
+            final String command = "bash " + Paths.get(testPlanLocation,TestGridConstants.INFRA_SCRIPT)
```

Comment:
Who provides the INFRA_SCRIPT/infra.sh? In AWS case, the cloudformation template yamls are taken via git using remoteRepository url + file path.

In this case, where should this infra.sh reside in? Is it under testgrid repo like this file: https://github.com/wso2/testgrid/blob/master/jobs/intg-test-resources/wso2-intg-test-cfn.yaml
or some other location?

This is a critical file we need to review as well.

Reply:
infra.sh,deploy.sh,destroy.sh all these scripts are not user provided. These scripts are testgrid provided scripts. It should come under the testgrid repo

Reply:
Ok, shall we add those into this PR itself?